src/test/java/seedu/weme/testutil/MemeUtil.java

```diff
@@ -60,17 +63,27 @@ public static String getEditMemeDescriptorDetails(EditMemeDescriptor descriptor)
     }
 
     /**
-     * Similar to {@link seedu.weme.model.util.MemeUtil#copyMeme(Meme, Path)}, except this method deletes the copied
-     * image file immediately afterwards.
-     *
-     * @param toCopy       the {@code Meme} to copy
-     * @param memeLocation the meme image location
-     * @return a new {@code Meme} with the new {@code ImagePath}.
+     * Returns the SHA-1 hash of a given file as a String.
      */
-    public static Meme generateCopiedMeme(Meme toCopy, Path memeLocation) throws IOException {
-        Meme copied = copyMeme(toCopy, memeLocation);
-        Files.delete(copied.getFilePath().getFilePath());
-        return copied;
+    public static String hash(Path file) {
+        try {
+            MessageDigest digest = MessageDigest.getInstance("SHA-1");
+            byte[] content = Files.readAllBytes(file);
+            return StringUtil.byteArrayToHex(digest.digest(content));
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+            throw new RuntimeException();
+        } catch (IOException e) {
+            throw new IllegalArgumentException(String.format(MESSAGE_READ_FILE_FAILURE, file.toString()));
+        }
+    }
+
+    /**
+     * Returns true if {@code firstMeme} and {@code secondMeme} supplied have the same SHA-1 hash,
+     * and therefore the same contents.
+     */
+    public static boolean isSameMemeImage(Meme firstMeme, Meme secondMeme) {
+        return hash(firstMeme.getFilePath().getFilePath()).equals(hash(secondMeme.getFilePath().getFilePath()));
```

Comment:
I was just thinking if it should be .getImagePath().getFilePath(). I'll change it in my own branch.

Reply:
Agree